tianshou/trainer/base.py

```diff
@@ -384,17 +398,27 @@ def test_step(self) -> tuple[CollectStats, bool]:
         )
         assert test_stat.returns_stat is not None  # for mypy
         rew, rew_std = test_stat.returns_stat.mean, test_stat.returns_stat.std
-        if self.best_epoch < 0 or self.best_reward < rew:
+        score = self.compute_score_fn(test_stat)
+        if self.best_epoch < 0 or self.best_score < score:
+            self.best_score = score
             self.best_epoch = self.epoch
             self.best_reward = float(rew)
             self.best_reward_std = rew_std
             if self.save_best_fn:
                 self.save_best_fn(self.policy)
-        log_msg = (
-            f"Epoch #{self.epoch}: test_reward: {rew:.6f} ± {rew_std:.6f},"
-            f" best_reward: {self.best_reward:.6f} ± "
-            f"{self.best_reward_std:.6f} in #{self.best_epoch}"
-        )
+        if score != rew:
```

Comment:
Personally, I'm not a fan of duplicated log message contents, but this is just FYI. I would have done something like 
```
score_info_addition = "" if score != rew else f", score: {score: .6f},"
```
and used it in `log_msg`. Feel free to leave as is or to adjust if you agree with me

Reply:
Perfect, thanks! Merged it